modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/spring/BeanDefinitionFactory.java

```diff
@@ -149,23 +150,24 @@ public void resolveEagerCreationObjects(ComponentModel parentComponentModel, Com
    * @param oldParsingMechanism a function to execute the old parsing mechanism if required by children {@code ComponentModel}s
    * @return the {@code BeanDefinition} of the component model.
    */
-  public BeanDefinition resolveComponentRecursively(ComponentModel parentComponentModel, ComponentModel componentModel,
+  public BeanDefinition resolveComponentRecursively(SpringComponentModel parentComponentModel,
+                                                    SpringComponentModel componentModel,
                                                     BeanDefinitionRegistry registry,
                                                     BiConsumer<ComponentModel, BeanDefinitionRegistry> componentModelPostProcessor,
                                                     BiFunction<Element, BeanDefinition, Either<BeanDefinition, BeanReference>> oldParsingMechanism) {
     List<ComponentModel> innerComponents = componentModel.getInnerComponents();
     if (!innerComponents.isEmpty()) {
       for (ComponentModel innerComponent : innerComponents) {
         if (innerComponent.isEnabled()) {
-          resolveComponentRecursively(componentModel, innerComponent, registry, componentModelPostProcessor,
-                                      oldParsingMechanism);
+          resolveComponentRecursively(componentModel, (SpringComponentModel) innerComponent, registry,
```

Comment:
Should not check that the component is of the expected class before casting it? (There are many places where this occurs)